DigitalLearningSolutions.Data.Tests/DataServices/GroupsDataServiceTests.cs

```diff
@@ -1,294 +1,344 @@
+
+                // When
+                groupsDataService.RemoveRelatedProgressRecordsForGroupDelegate(8, delegateId, removedDate);
+                var progressFields = await connection.GetProgressRemovedFields(285172);
+
+                // Then
+                progressFields.Item1.Should().Be(0);
+                progressFields.Item2.Should().BeNull();
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [Test]
+        public void GetRelatedProgressIdForGroupDelegate_returns_expected_value()
+        {
+            // Given
+            const int delegateId = 245969;
+
+            // When
+            var result = groupsDataService.GetRelatedProgressIdForGroupDelegate(5, delegateId);
+
+            // Then
+            result.Should().Be(285146);
+        }
+
+        [Test]
+        public void GetRelatedProgressIdForGroupDelegate_returns_null_with_no_record()
+        {
+            // Given
+            const int delegateId = 2;
+
+            // When
+            var result = groupsDataService.GetRelatedProgressIdForGroupDelegate(1, delegateId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void UpdateGroupDescription_updates_record()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                const int centerId = 101;
+                const int groupId = 5;
+                const string expectedDescription = "Test group description";
+
+                // When
+                groupsDataService.IsGroupDescriptionUpdated(groupId, centerId, expectedDescription);
+
+                // Then
+                var result = groupsDataService.GetGroupAtCentreById(groupId, centerId);
+                result?.GroupDescription.Should().Be(expectedDescription);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [Test]
+        public void UpdateGroupDescription_with_incorrect_centreId_does_not_update_record()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                const int incorrectCentreId = 107;
+                const int groupId = 5;
+                const string expectedDescription = "Test group description";
+
+                // When
+                groupsDataService.IsGroupDescriptionUpdated(groupId, incorrectCentreId, expectedDescription);
+
+                //Then
+                var result = groupsDataService.GetGroupAtCentreById(groupId, incorrectCentreId);
+                result?.GroupDescription.Should().NotBe(expectedDescription);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
```

Comment:
The usage of the transactions is now correct, however you haven't addressed the fact that we aren't testing the correct thing here. The test method name also needs updating to match the new method name.

This test should have 2 assertions:

1. checking the return type of the method (false for this test)
2. the value in the database has NOT been updated. To do this we will need a helper method to query the actual value based on only the groupId. 

You will also need to fix the database value on test that got updated by the previous version of these tests without the transaction (it should be NULL) otherwise we won't be able to tell that these tests are failing.

Reply:
The test database should be restored to the state it was before the incorrect test was run, i.e. having the description as NULL. We need to keep it as consistent as we can so that if someone else were to write a set of tests relying on that record we don't get unexpected test errors due to differences in the database.

If you want the value to be there for a test for some reason, it then becomes something you need to write a script for so that everyone else can make the same change to their test databases.

Reply:
Restored test database and tested all test cases. All are passed